dashboard/app/models/script.rb

```diff
@@ -1631,4 +1633,100 @@ def all_descendant_levels
     sublevels = levels.map(&:all_descendant_levels).flatten
     levels + sublevels
   end
+
+  def serialize_seeding_json
+    eager_loaded_script_levels = ScriptLevel.includes(:levels).where(script_id: id)
+    eager_loaded_levels = eager_loaded_script_levels.map(&:levels).flatten
```

Comment:
preloading levels 100% seems like the right move. I haven't thought through whether/why preloading script levels makes sense, but it seems reasonable if it makes things work. can you offer any insight as to why script levels need to be preloaded but other types (e.g. lessons) do not? is it just because there are so many more script levels?

 possibly useful list of different ways to load association data: https://blog.bigbinary.com/2013/07/01/preload-vs-eager-load-vs-joins-vs-includes.html (maybe you named these when you were trying `eager_load`)

Reply:
If I'm remembering correctly, the valuable part for performance is `.includes(:levels)` for the script levels, since that makes finding the level keys for each script level fast when serializing the script levels. Then, since we have the levels loaded already, might as well save a query and reuse them for the list of levels.

I'm not sure this optimization is needed, but it doesn't seem overly complex, so I'm ok with keeping it for now. I added a comment and renamed the variables. LMK if you'd rather get rid of it.